### ISSUE TITLE
 Cannot convert argument to a ByteString

### ISSUE THREAD

**Issue by benoit74** [webrecorder/browsertrix-crawler#569]
We face a strange error when crawling a website: `Cannot convert argument to a ByteString because the character at index 63 has a value of 305 which is greater than 255.`

This happened during a crawl of `https://www.hra-news.org` website, and occurred on `https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg` page (which is not really a page btw, but this has probably been linked as a page somewhere on the website ...).

Command used for repro:

```
docker run -v $PWD/output:/output --name zimit2 --rm  webrecorder/browsertrix-crawler:1.1.1 crawl --failOnFailedSeed --behaviors "autoplay,autofetch,autoscroll" --url "https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg" --scopeType "host" --mobileDevice "Pixel 2" --cwd /output --combineWARC --depth 1
```

Detailed logs of the repro:
```
{"timestamp":"2024-05-13T06:18:19.493Z","logLevel":"info","context":"general","message":"Browsertrix-Crawler 1.1.1 (with warcio.js 2.2.1)","details":{}}
{"timestamp":"2024-05-13T06:18:19.494Z","logLevel":"info","context":"general","message":"Seeds","details":[{"url":"https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg","scopeType":"host","include":["/^https?:\\/\\/www\\.hra\\-news\\.org\\//"],"exclude":[],"allowHash":false,"depth":-1,"sitemap":null,"maxExtraHops":0,"maxDepth":1}]}
{"timestamp":"2024-05-13T06:18:19.915Z","logLevel":"info","context":"worker","message":"Creating 1 workers","details":{}}
{"timestamp":"2024-05-13T06:18:19.915Z","logLevel":"info","context":"worker","message":"Worker starting","details":{"workerid":0}}
{"timestamp":"2024-05-13T06:18:20.020Z","logLevel":"info","context":"worker","message":"Starting page","details":{"workerid":0,"page":"https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg"}}
{"timestamp":"2024-05-13T06:18:20.021Z","logLevel":"info","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":0,"total":1,"pending":1,"failed":0,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2024-05-13T06:18:19.916Z\",\"extraHops\":0,\"url\":\"https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg\",\"added\":\"2024-05-13T06:18:19.522Z\",\"depth\":0}"]}}
{"timestamp":"2024-05-13T06:18:22.505Z","logLevel":"info","context":"general","message":"Awaiting page load","details":{"page":"https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg","workerid":0}}
node:internal/deps/undici/undici:1834
          throw new TypeError(
                ^

TypeError: Cannot convert argument to a ByteString because the character at index 63 has a value of 305 which is greater than 255.
    at webidl.converters.ByteString (node:internal/deps/undici/undici:1834:17)
    at Object.record<ByteString, ByteString> (node:internal/deps/undici/undici:1743:32)
    at webidl.converters.HeadersInit (node:internal/deps/undici/undici:2283:67)
    at new Headers (node:internal/deps/undici/undici:2080:36)
    at RequestResponseInfo.getMimeType (file:///app/dist/util/reqresp.js:196:25)
    at Recorder.addPageRecord (file:///app/dist/util/recorder.js:448:34)
    at Recorder.serializeToWARC (file:///app/dist/util/recorder.js:669:14)
    at Recorder.handleRedirectResponse (file:///app/dist/util/recorder.js:197:14)
    at Recorder.handleRequestWillBeSent (file:///app/dist/util/recorder.js:160:18)
    at file:///app/dist/util/recorder.js:71:62

Node.js v20.11.1
```

Original logs (end only obviously):
```
{"timestamp":"2024-05-08T21:59:51.589Z","logLevel":"info","context":"pageStatus","message":"Page Finished","details":{"loadState":4,"page":"https://www.hra-news.org/tag/%d8%b2%d9%86%d8%af%d8%a7%d9%86%db%8c-%d8%a8%db%8c%d9%85%d8%a7%d8%b1/page/15/","workerid":0}}
{"timestamp":"2024-05-08T21:59:51.615Z","logLevel":"info","context":"worker","message":"Starting page","details":{"workerid":0,"page":"https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg"}}
{"timestamp":"2024-05-08T21:59:51.617Z","logLevel":"info","context":"crawlStatus","message":"Crawl statistics","details":{"crawled":85574,"total":162856,"pending":1,"failed":1,"limit":{"max":0,"hit":false},"pendingPages":["{\"seedId\":0,\"started\":\"2024-05-08T21:59:51.614Z\",\"extraHops\":0,\"url\":\"https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg\",\"added\":\"2024-05-04T12:02:58.068Z\",\"depth\":4}"]}}
{"timestamp":"2024-05-08T21:59:54.193Z","logLevel":"info","context":"general","message":"Awaiting page load","details":{"page":"https://www.hra-news.org/fa/wp-content/uploads/2016/11/Reza-Alijan%C4%B1.jpg","workerid":0}}
node:internal/deps/undici/undici:1834
          throw new TypeError(
                ^

TypeError: Cannot convert argument to a ByteString because the character at index 63 has a value of 305 which is greater than 255.
    at webidl.converters.ByteString (node:internal/deps/undici/undici:1834:17)
    at Object.record<ByteString, ByteString> (node:internal/deps/undici/undici:1743:32)
    at webidl.converters.HeadersInit (node:internal/deps/undici/undici:2283:67)
    at new Headers (node:internal/deps/undici/undici:2080:36)
    at RequestResponseInfo.getMimeType (file:///app/dist/util/reqresp.js:196:25)
    at Recorder.addPageRecord (file:///app/dist/util/recorder.js:448:34)
    at Recorder.serializeToWARC (file:///app/dist/util/recorder.js:669:14)
    at Recorder.handleRedirectResponse (file:///app/dist/util/recorder.js:197:14)
    at Recorder.handleRequestWillBeSent (file:///app/dist/util/recorder.js:160:18)
    at file:///app/dist/util/recorder.js:71:62

Node.js v20.11.1
```

**Comment by benoit74** [comment]
Another user reported same issue on another web property : https://github.com/openzim/zimit/issues/296#issuecomment-2110263443

**Comment by ikreymer** [comment]
This is due to the location header containing non-ASCII values. Simplest option is to detect this and call `encodeURI` to encode these. Have a fix for this in #573.

**Comment by ikreymer** [comment]
Fixed in 1.1.2
